backend/src/add_post.rs

```
@@ -90,26 +90,56 @@ impl NewPost {
                     content: content.unwrap_or_default(),
                     media_url: media_url.unwrap_or_default(),
                     time_ago: time_ago(created_on),
-                    date: common::datetime_to_date_string(&created_on)
+                    date: common::datetime_to_date_string(&created_on),
                 },
                 None => PostWithTime {
                     content: "".to_string(),
                     media_url: "".to_string(),
                     time_ago: "".to_string(),
-                    date: "".to_string()
+                    date: "".to_string(),
                 },
             };
 
+        let post_count = post_count_by_user_id(conn, user_id)?;
+
         Ok(Output {
             post_id,
             post_content: self.post_content,
             media_url: self.media_url,
             created_on: self.created_on.format("%Y-%m-%d %H:%M:%S").to_string(),
             random_post,
+            post_count,
         })
     }
 }
 
+/// Returns the number of posts by the user_id whose subscription is free trial
+fn post_count_by_user_id(conn: &mut ft_sdk::Connection, user_id: i64) -> Option<i64> {
```

The signature is wrong. You are eating errors up. If you ever do any operation that can return error (eg db is down etc), you must not eat the error. Change the return type to `Result<Option<i64>, PostCountError>`.